packages/gatsby-admin/src/utils/use-develop-logs.ts

```diff
@@ -0,0 +1,65 @@
+import React from "react"
+import io from "socket.io-client"
+import { useServices } from "../components/services-provider"
+
+type RestartState = "idle" | "needs-restart" | "is-restarting"
+
+type RestartFn = () => void
+
+function useDevelopState(): [RestartState, RestartFn] {
+  const [state, setState] = React.useState<RestartState>(`idle`)
+  const [socket, setSocket] = React.useState<ReturnType<typeof io> | null>(null)
+  const [restartFn, setRestartFn] = React.useState<RestartFn>(() => null)
+
+  const services = useServices()
+  React.useEffect(() => {
+    if (!services.developstatusserver) return
+
+    setSocket(
+      io(
+        `${window.location.protocol}//${window.location.hostname}:${services.developstatusserver.port}`
+      )
+    )
+  }, [services])
+
+  React.useEffect((): Function => {
+    // NOTE(@mxstbr): Have to return a function here, otherwise eslint complains.
+    if (!socket) return (): void => {}
+
+    setRestartFn(
+      (): RestartFn => (): void => {
+        setState(`is-restarting`)
+        socket.emit(`develop:restart`)
+      }
+    )
```

Comment:
Maybe I'm lacking context so this comment might not be relevant:  WYT of extracting the `restartFn` outside the `useEffect`?

Something like:

```javascript
const restartFn = () => {
    if (socket) {
        setState(`is-restarting`)
        socket.emit(`develop:restart`)
     }
}
```

This would help get rid off the whole `useEffect` and avoiding storing a function in state 😊